### PR TITLE
Added env variable to ignore ./ and ../ in filepath completion

### PR DESCRIPTION
**Pull request by Baileymnm1999**
## For community
**Added:**

* Added environment variable `$IGNORE_DOTS` to ignore ./ and ../ in file path completions.

This addresses issue  #4213 

⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

**Comment by anki-code** [comment]
Awesome! Thanks! Please add the default value i.e. `env.get("IGNORE_DOTS", False)`. In some test cases it's good to avoid the value error.

**Review by gforsyth — CHANGES_REQUESTED** [review]
Thanks for putting this in @Baileymnm1999 !

I am actually just going to paste out @anki-code 's review here, because he hit all the points I was going to raise 😅 

>Please add the default value i.e. env.get("IGNORE_DOTS", False). In some test cases it's good to avoid the value error.
> It will be cool to have the name of the variable in the news.
> And also Gil will ask you about adding the env variable to the white list. 

Regarding the last point -- it should be defined where @anki-code indicated and it should have a default value.  Usually I would have the default value reflect current behavior, but in this case I'm leaning towards dropping the dots by default as offering tab completion for them doesn't really offer any benefits in terms of saved keystrokes.

**Comment by anki-code** [comment]
> I'm leaning towards dropping the dots by default as offering tab completion for them doesn't really offer any benefits in terms of saved keystrokes.

May be just drop them without env variable :)

**Comment by gforsyth** [comment]
Hmm, that's a good point @anki-code.

How about we don't include any `.` results unless a user has already typed one?

e.g.:

in an empty directory

`cd <tab>` -> no completions

`cd .<tab>` -> `./` and `../`

@Baileymnm1999 -- would you be up for tackling this?

**Comment by Baileymnm1999** [comment]
@gforsyth what do you think of changing the environment variable to `COMPLETE_DOTS = {always, never, matching}` or something similar with it defaulting to matching?

**Review by Baileymnm1999 — COMMENTED** [review]
Here are the changes to address all the coments

**Comment by Baileymnm1999** [comment]
I am not sure why qa checks are failing, pep8 linter doesn't show me anything

**Comment by anki-code** [comment]
You can click on Details and get the error in tests:
```
tests/completers/test_dir_completers.py
# ...
    def _dots(prefix):
>       complete_dots = builtins.__xonsh__.env.get("COMPLETE_DOTS").lower()
E       AttributeError: 'NoneType' object has no attribute 'lower'

/usr/share/miniconda/envs/xonsh-test/lib/python3.7/site-packages/xonsh/completers/path.py:127: AttributeError
```
Then you can run the test locally:
```python
pytest tests/completers/test_dir_completers.py
```
The error is about you should set the default value for the env variable i.e. `get("COMPLETE_DOTS", "matching")`.

**Comment by Baileymnm1999** [comment]
I think this change affects expected functionality in the unit tests. Should they be changed?

**Comment by Baileymnm1999** [comment]
Sorry, for the difficulties and thanks for the patience. This is my first ever attempted pr on github.

**Comment by anki-code** [comment]
Okay, we're very close to success. 

Check the tests details. After short look I see two errors:

1. The syntax in the news file is wrong for RST. I recommed to simplify the news: `Added environment variable COMPLETE_DOTS to specify how current and previous directories should be tab completed in cd.` (without back quotes)

2. The test `tests/completers/test_base_completer.py` should be fixed.

**Comment by anki-code** [comment]
Hi @Baileymnm1999 ! Please pull the code from the main branch the fix for one test that failed before. And it will be great to continue the work on this feature! I see many thumbs up on the first message, it's great!

File: news/ignore_dots.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
+**Added:**
+
+* Added environment variable `$COMPLETE_DOTS` to specify how current and previous directories should be tab completed in cd  ('./', '../'):
```
Comment (gforsyth) [inline news/ignore_dots.rst:3]:
```suggestion
* Added environment variable ``$COMPLETE_DOTS`` to specify how current and previous directories should be tab completed in cd  ('./', '../'):
```

rst is very picky about the number of backticks

File: xonsh/completers/path.py [hunk 2]
```diff
@@ -124,10 +124,16 @@ def _env(prefix):
 
 
 def _dots(prefix):
+    complete_dots = builtins.__xonsh__.env.get("COMPLETE_DOTS", "matching").lower()
+    if complete_dots == 'never':
```
Comment (gforsyth) [inline xonsh/completers/path.py:128]:
```suggestion
    if complete_dots == "never":
```

We use `black` to autoformat all python files and it prefers double-quotes to single (unless they're nested)

File: xonsh/completers/path.py [hunk 3]
```diff
@@ -124,10 +124,16 @@ def _env(prefix):
...
+        return ()
     slash = xt.get_sep()
     if slash == "\\":
         slash = ""
-    if prefix in {"", "."}:
+    prefixes = {"."}
+    if complete_dots == 'always':
```
Comment (gforsyth) [inline xonsh/completers/path.py:134]:
```suggestion
    if complete_dots == "always":
```

**Review by gforsyth — CHANGES_REQUESTED** [review]
Hey @Baileymnm1999 -- this looks good and it's very close!  I think just a few small formatting fixes and a merge from the `main` branch to fix a flaky test and you should be good to go.

**Comment by gforsyth** [comment]
> I think this change affects expected functionality in the unit tests. Should they be changed?

If the test expects the `.`, `..` to be in the results, it's ok to remove that from the expected results as that will reflect the new expected behavior.

**Pull request closed without merging**